src/Stratis.Bitcoin.Features.BlockStore/AddressIndexing/AddressIndexerOutpointsRepository.cs

```diff
@@ -95,6 +101,44 @@ public void SaveAllItems()
             }
         }
 
+        /// <summary>Persists rewind data into the repository.</summary>
+        /// <param name="rewindData">The data to be persisted.</param>
+        public void RecordRewindData(AddressIndexerRewindData rewindData)
+        {
+            lock (this.LockObject)
+            {
+                this.addressIndexerRewindData.Insert(rewindData);
+            }
+        }
+
+        /// <summary>Deletes rewind data that is more than maxReorg blocks old.</summary>
+        /// <param name="height">The threshold below which data will be deleted.</param>
+        public void PurgeOldRewindData(int height)
+        {
+            lock (this.LockObject)
+            {
+                // Generally there will only be one result here at most, as this should be getting called once per block.
+                foreach (AddressIndexerRewindData rewindData in this.addressIndexerRewindData.Find(Query.LT("BlockHeightIndex", height)))
```

Comment:
What if there is more than one record to delete? Perhaps do a check and then if more than 1 batch delete? Just worried about running into a scenario where we have to do 1000+ deletes?

Reply:
As discussed, LiteDb doesn't really have batch deletes.